### PR TITLE
Fix doxygen pirs898 compile and improve documentation

### PR DESCRIPTION
**Pull request by rtownson**
Removes some warnings from the pirs898 compilation.

Improves appearance of documentation for the egs++ library, and fixes a number of typos and display bugs. There's still work to be done on this front but it's a start.

### CODE REVIEW

**Comment by ftessier** [comment]
Thanks @rtownson! Can we squash the two commits, or should we keep them separated?

**Comment by rtownson** [comment]
Squashing the two is fine! By the way, the large Doxyfile overhaul in the first commit was done using doxygen -u, which updates the file to the installed version of doxygen (1.8.5).

**Comment by ftessier** [comment]
Thanks for pointint that out, I was wondering about all those changes! I am going to rebase this branch on release-2017, because I want to compile the egs++ documentation with all the changes to egs++ files on release-2017, and also fix a doxygen typo in `egs_stack_geometry.h`.

File: HEN_HOUSE/egs++/geometry/egs_gstack/egs_stack_geometry.h [hunk 1]
```diff
@@ -103,14 +103,14 @@ library = egs_gstack
 geometries = list of names of previously defined geometries
 tolerance = small floating number
 \endverbatim
-The tolerance key defines a small floating point number \f$\boundaryTolerance\f$.
```
Comment (ftessier) [inline HEN_HOUSE/egs++/geometry/egs_gstack/egs_stack_geometry.h]: Remove extra `\` characters which caused doxygen compilation to fail.

**Comment by mainegra** [comment]
This looks much nicer than the previous layout! @rtownson could the front page have the main title with the picture on the left and a 2X2 grid with the different sections? The main reason for this would be to keep all the relevant information visible.

**Comment by ftessier** [comment]
May I suggest we update the cover image too? Perhaps an ion chamber with clipping plane and particle tracks?

**Comment by ftessier** [comment]
Nice clean layout @rtownson, thanks! I changed the main page slightly, and changed "user code" to "application" in the doxy files.

**Pull request merged**